db/view/view_update_backlog.hh

```diff
@@ -23,12 +27,29 @@ namespace db::view {
  * backlog - how much view hints are consuming. The size of a backlog is relative
  * to its maximum size.
  */
-struct update_backlog {
-    size_t current;
-    size_t max;
+class update_backlog {
+    size_t _current;
+    size_t _max;
+
+public:
+    update_backlog(size_t current, size_t max)
+            : _current(current), _max(max) {
+        if (max == 0) {
+            on_internal_error(vlogger, "Constructing invalid update_backlog (max = 0)");
+        }
```

Comment:
Sorry for changing my mind, but I think we should drop the on_internal_error after all. Because we might use some data that comes from a different node, it's better not to assume that it's correct. The code that creates the update_backlog might not be prepared to handle the exception. Even if it is not possible right now to create a backlog with zero maximum, it's harder to say this about the other versions we might rolling upgrade from. I realized the last one just now.

It's safer to set the backlog to some safe value instead of throwing, e.g. we can set max to `std::numeric_limits<size_t>::max()` if the provided value is zero.

Reply:
Ok, I changed the behavior so that we make an empty backlog if we try to construct an invalid one (the `current` is also meaningless if the size is 0 so I set it to 0 as well)